content/altinn-studio/getting-started/app-dev-course/modul7/_index.nb.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Modul 7: Legge til dynamikk"
+description: 
+linktitle: "Modul 7: Legge til dynamikk"
+tags: [apper, kurs, dynamikk]
+weight: 70
+toc: true
+---
+ 
+I denne modulen skal vi videreutvikle applikasjonen ved å legge til dynamisk skjuling av sider og komponenter. Dette gjør 
+at du kan styre hvilke deler av appen som vises, basert på brukerens svar.
+
+## Temaer som dekkes i denne modulen
+- Vise og skjule sider ved hjelp av logiske uttrykk
+- Vise og skjule komponenter ved hjelp av logiske uttrykk
+
+## Krav fra Sogndal kommune
+Sogndal kommune har to krav for dynamikk.
+
+### Krav 1: Stopp brukere som ikke oppfyller kravene tidlig
+En bruker som ikke oppfyller kravene for skjemaet skal stoppes så tidlig som mulig i arbeidsflyten. 
+Hvis brukeren ikke bekrefter at skjemaet er relevant, skal de sendes til en ny forklaringsside. Hvis de bekrefter relevansen,
+skal de sendes til datainnsamlingssidene.
+
+### Krav 2: Vis stillingsutlysninger for IKT-bransjen
+Om en bruker velger **IKT (data/it)** under bransje skal det vises en tekst med en lenke til en av deres stillingsutlysninger.
+
+## Implementer dynamisk visning og skjuling av sider
+For å oppfylle krav 1 fra Sogndal kommune, må vi opprette en ny side som skal vises hvis appen ikke er relevant for brukeren. 
+Vi legger også til en funksjon der brukeren kan bekrefte relevansen, og basert på dette valget vil vi dynamisk skjule eller vise sider ved hjelp av logiske uttrykk.
+
+### Legg til ny side
+Vi legger til en ny side som skal vises hvis brukeren ikke bekrefter at de oppfyller kravene for appen.
+1. Klikk på **Utforming**.
+2. Klikk på **Legg til ny side** og gi den navnet ```ikke-for-deg``` 
+3. Dra inn en komponent av typen **Avsnitt**.
+4. Klikk på **Ledetekst** under **Tekst**-menyen til komponenten.
+5. Kopier inn følgende tekst: 
+    ```md
+    Dette skjemaet er ikke for deg.
+    Se en oversikt over andre tilbud i kommunen [her](https://www.sogndal.kommune.no/).
+    ```
+Appen skal nå bestå av fire sider.
+
+!["Fire sider"](fire_sider.png)
+
+### Oppdater datamodellen
+For at appen skal kunne vurdere om skjemaet er relevant for brukeren, må vi oppdatere datamodellen.
+1. Klikk på **Datamodell**.
+2. Klikk på **Legg til**.
+3. Klikk på **Ja/nei**.
+4. Endre navnet på feltet til ```KanBrukeSkjema```.
+5. Klikk **Generer modeller**.
+
+Datamodellen skal nå se slik ut:
+
+!["Datamodell med KanBrukeSkjema"](datamodell.png)
+
+### Legg til avkrysningsbokser
+Vi legger til avkrysningsbokser på infosiden så brukeren kan bekrefte at skjemaet er relevant for dem.
```

Comment:
```suggestion
Vi legger til avkrysningsbokser på infosiden så brukeren kan bekrefte at skjemaet er relevant for dem.
```Vi legger til avmerkingsbokser på infosiden, så brukerne kan bekrefte at skjemaet er relevant for dem.